source/persistence/MessageController.cpp

```diff
@@ -368,3 +373,53 @@ void MessageController::printSqlError()
     qDebug() << this->lastError().driverText();
     qDebug() << this->lastError().text();
 }
+
+
+void MessageController::removeMessagesFromJid(const QString& jid)
+{
+    // First parse the list of messages with attachments
+
+    QSqlQuery query(*(database_->getPointer()));
+
+    if (! query.exec("SELECT " + Database::sqlMsgMessage_ + " FROM " + Database::sqlMsgName_
+                     + " WHERE " + Database::sqlJid_ + " = \"" + jid + "\"" + "AND NOT " + Database::sqlMsgType_ + " = \"txt\""))
+    {
+        qDebug() << query.lastError().databaseText();
+        qDebug() << query.lastError().driverText();
+        qDebug() << query.lastError().text();
+    }
+    else
+    {
+        while (query.next())
+        {
+            QUrl url = query.value(0).toUrl();
+
+            if(url.isValid() && url.scheme().length() > 0)
+            {
+                QString hash = CryptoHelper::getHashOfString(url.toString(), true);
+                QString pathAndFile = System::getAttachmentPath() + QDir::separator() + hash;
+                QFile attachFile(pathAndFile);
+
+                attachFile.remove();
+            }
+        }
+    }
+
+    // Then remove messages
+
+    if (! query.exec("DELETE FROM " + Database::sqlMsgName_
+                     + " WHERE " + Database::sqlJid_ + " = \"" + jid + "\"" ))
+    {
+        qDebug() << query.lastError().databaseText();
+        qDebug() << query.lastError().driverText();
+        qDebug() << query.lastError().text();
+    }
+    else
+    {
+        // update the model with the changes of the database
+        if (select() != true)
+        {
+            qDebug() << "error on select in MessageController::removeMessagesFromJid";
+        }
+    }
+}
```

Comment:
Hm, this removes all the messages for a given JID. I was expecting to be able to remove one specific message within a  conversation. But maybe I did get the original issue request wrong. I had to read it again...